### PR TITLE
Adding accounts package

### PR DESCRIPTION
**Pull request by nathanathan**
Connects to #9

This adds user registration and sign-in pages. It also adds a header with sign-in/sign-out links. I added forks of the [useraccounts](https://atmospherejs.com/useraccounts/core) packages to the repository because the current versions of them on Atmosphere do not yet work with the flow router. However, Amy and I have been discussing removing the forks from this repository and putting them on Atmosphere instead.

### CODE REVIEW

File: packages/accounts/useraccounts_templates.coffee [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Template.signIn.events
+  'click #at-signUp' : ->
+    FlowRouter.go "/signUp"
```
Comment (aslagle) [inline packages/accounts/useraccounts_templates.coffee]: This package shouldn't depend on FlowRouter directly. We could add something to the 'path' package and rename it.

File: packages/accounts/user_publication.coffee [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Meteor.publish "userInfo", (userId) ->
+  Meteor.users.find {_id: userId}, {profile: 1}
```
Comment (aslagle) [inline packages/accounts/user_publication.coffee]: This is for the profile page, right? I think we can leave it out for now.

File: packages/layout/header.jade [hunk 3]
```diff
@@ -1,2 +1,26 @@
...
+      a.dropdown-toggle.admin-settings(href="#" data-toggle="dropdown" aria-expanded="false")
+        i.fa.fa-gear
+        span Settings
+      ul.dropdown-menu(role="menu")
+        +accountsHeaderButtons
+  else
+    li
+      a(href="/signIn" class="sign-in navbar-btn")
+        span Sign In 
+        i.fa.fa-sign-in
```
Comment (aslagle) [inline packages/layout/header.jade]: The sign-in link should part of the accounts header buttons template or at least the accounts package. I don't think we need the settings dropdown at this point, we haven't decided anything about the mantle header yet.

File: packages/router/router.coffee [hunk 4]
```diff
@@ -4,3 +4,15 @@ FlowRouter.route '/',
     FlowLayout.render 'layout',
       main: 'splashPage'
       params: {}
+
+FlowRouter.route '/signIn',
+  name: 'signIn'
+  action: () ->
+    FlowLayout.render 'layout',
+      main: 'signIn'
```
Comment (aslagle) [inline packages/router/router.coffee]: I'd like to see what it would look like to render the sign in template without changing the route.

**Comment by aslagle** [comment]
If we need to leave the useraccounts package in the repository for now, I guess it's ok, but since we didn't make changes, maybe we could preserve the package name? Leave it as useraccounts:core instead of eha:useraccounts-core. Then maybe we wouldn't need to have the other bootstrap package as well.

**Comment by nathanathan** [comment]
@aslagle I made the changes you asked for and I created an example of rendering the signIn template without changing the route on this branch: https://github.com/ecohealthalliance/mantle/tree/no-route-change-signin

**Comment by nathanathan** [comment]
@aslagle I implemented your idea to put the sign-in/registration forms in a modal in this commit: 9822810

File: packages/accounts/package.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Package.onUse(function(api) {
+  api.versionsFrom('1.1.0.2');
+  api.use('templating');
+  api.use('coffeescript');
+  api.use('underscore');
+  api.use('reactive-var');
+  api.use('mquandalle:jade@0.4.3');
+  
+  api.use('mantle:path');
```
Comment (aslagle) [inline packages/accounts/package.js]: I don't think we need the dependency on mantle:path anymore

File: packages/layout/header.jade [hunk 6]
```diff
@@ -1,2 +1,11 @@
-template(name='header')
-  p HEADER
\ No newline at end of file
+template(name="header")
+  header
+    nav.navbar
+      .fluid-container
+        .container
```
Comment (aslagle) [inline packages/layout/header.jade]: what is the reason for having both .fluid-container and .container? @jgoley?

File: packages/path/package.js [hunk 7]
```diff
@@ -10,7 +10,11 @@ Package.onUse(function(api) {
   api.use('coffeescript');
   api.use('templating');
   api.use('meteorhacks:flow-router@1.9.0');
+  
   api.addFiles('helpers.coffee', 'client');
+  api.addFiles('path.coffee', 'client');
+  
+  api.export('Path', 'client');
```
Comment (aslagle) [inline packages/path/package.js]: I think we can remove this until we actually need it.

**Comment by aslagle** [comment]
Thanks, looks great except for those minor comments. I like this approach so far.

File: packages/layout/header.jade (hunk 6 above)
Comment (jgoley) [inline packages/layout/header.jade]: I established that structure when I was learning bootstrap. ```.fluid-container``` adds padding on the left and right which is necessary at the smallest breakpoint. I'll see how we can remove the .fluid-container and keep the padding.

Comment (aslagle) [inline packages/layout/header.jade]: Cool let's just remove it for now.

Comment (jgoley) [inline packages/layout/header.jade]: Removed. The padding seems okay so i probably corrected that issue in EIDR and maybe Rana with custom styles. I'll fix it in the others.

**Pull request merged**